programs/castle-vault/src/adapters/solend.rs

```diff
@@ -241,86 +263,113 @@ pub fn refresh_reserve<'info>(
     // Lending market authority (PDA)
+    /// CHECK: safe
     pub lending_market_authority: AccountInfo<'info>,
     // Transfer authority for accounts 1 and 2
+    /// CHECK: safe
     pub transfer_authority: AccountInfo<'info>,
     // Clock
+    /// CHECK: safe
     pub clock: AccountInfo<'info>,
     // Token program ID
+    /// CHECK: safe
     pub token_program_id: AccountInfo<'info>,
 }
 
 #[derive(Accounts)]
 pub struct RedeemReserveCollateral<'info> {
     // Lending program
+    /// CHECK: safe
     pub lending_program: AccountInfo<'info>,
     // Source token account for reserve collateral token
+    /// CHECK: safe
     pub source_collateral: AccountInfo<'info>,
     // Destination liquidity token account
+    /// CHECK: safe
     pub destination_liquidity: AccountInfo<'info>,
     // Refreshed reserve account
+    /// CHECK: safe
     pub reserve: AccountInfo<'info>,
     // Reserve collateral mint account
+    /// CHECK: safe
     pub reserve_collateral_mint: AccountInfo<'info>,
     // Reserve liquidity supply SPL Token account.
+    /// CHECK: safe
     pub reserve_liquidity_supply: AccountInfo<'info>,
     // Lending market account
+    /// CHECK: safe
     pub lending_market: AccountInfo<'info>,
     // Lending market authority - PDA
+    /// CHECK: safe
     pub lending_market_authority: AccountInfo<'info>,
     // User transfer authority
+    /// CHECK: safe
     pub transfer_authority: AccountInfo<'info>,
     // Clock
+    /// CHECK: safe
     pub clock: AccountInfo<'info>,
     // Token program ID
+    /// CHECK: safe
     pub token_program_id: AccountInfo<'info>,
 }
 
 #[derive(Accounts)]
 pub struct RefreshReserve<'info> {
     // Lending program
+    /// CHECK: safe
     pub lending_program: AccountInfo<'info>,
     // Reserve account
+    /// CHECK: safe
     pub reserve: AccountInfo<'info>,
     // Pyth reserve liquidity oracle
     // Must be the pyth price account specified in InitReserve
+    /// CHECK: safe
     pub pyth_reserve_liquidity_oracle: AccountInfo<'info>,
     // Switchboard Reserve liquidity oracle account
     // Must be the switchboard price account specified in InitReserve
+    /// CHECK: safe
     pub switchboard_reserve_liquidity_oracle: AccountInfo<'info>,
     // Clock
+    /// CHECK: safe
     pub clock: AccountInfo<'info>,
 }
 
 #[derive(Clone)]
 pub struct SolendReserve(Reserve);
 
 impl anchor_lang::AccountDeserialize for SolendReserve {
-    fn try_deserialize(buf: &mut &[u8]) -> Result<Self, ProgramError> {
+    fn try_deserialize(buf: &mut &[u8]) -> Result<Self> {
         SolendReserve::try_deserialize_unchecked(buf)
     }
 
-    fn try_deserialize_unchecked(buf: &mut &[u8]) -> Result<Self, ProgramError> {
-        <Reserve as solana_program::program_pack::Pack>::unpack(buf).map(SolendReserve)
+    fn try_deserialize_unchecked(buf: &mut &[u8]) -> Result<Self> {
+        Ok(<Reserve as solana_program::program_pack::Pack>::unpack(buf).map(SolendReserve)?)
```

Comment:
Why does this need to be wrapped in an Ok?

Reply:
alternate way to handle the conversion from ProgramError to Error